src/sdk/pynni/nni/nas/pytorch/pdarts/mutator.py

```diff
@@ -25,13 +27,13 @@ def __init__(self, model, pdarts_epoch_index, pdarts_num_to_drop, switches={}):
         for mutable in self.mutables:
             if isinstance(mutable, LayerChoice):
 
-                switches = self.switches.get(mutable.key, [True for j in range(mutable.length)])
+                switches = self.switches.get(mutable.key, [True for j in range(mutable.length+1)])
+                switches.requires_grad = False
+                choices = self.choices[mutable.key]
 
-                for index in range(len(switches)-1, -1, -1):
+                for index in range(len(switches)-2, -1, -1):
```

Comment:
better to add comment to explain why `-2`

Reply:
added

Reply:
new logic is simpler, so no -2 is here.